pom.xml

```diff
@@ -126,7 +126,7 @@
       <dependency>
         <groupId>org.jenkins-ci.plugins.workflow</groupId>
         <artifactId>workflow-step-api</artifactId>
-        <version>2.24</version>
+        <version>622.vb_8e7c15b_c95a_</version>
```

Comment:
@olamy thus dependency should have been removed from there - it is in the BOM - so either we need a newer bom, or something else is broken.

Reply:
```


[WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 paths to dependency are:
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-api:1.9.2
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-job:2.41
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-api:2.47
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-cps:2.94
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:credentials-binding:1.27
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:git:4.9.0
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.24
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:htmlpublisher:1.29
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:622.vb_8e7c15b_c95a_
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:junit:1.53
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.24
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-basic-steps:2.24 [test]
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 [test] (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 [test]
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-durable-task-step:2.40 [test]
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 [test] (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23 [test]
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:token-macro:2.15 [test]
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 [test] (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23 [test]
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:pipeline-build-step:2.15 [test]
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 [test] (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23 [test]
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:pipeline-utility-steps:2.10.0 [test]
    +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 [test] (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23 [test]
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-job:2.41
    +-org.jenkins-ci.plugins.workflow:workflow-support:3.8 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-support:3.8
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins.workflow:workflow-cps:2.94
    +-org.jenkins-ci.plugins.workflow:workflow-scm-step:2.13 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-scm-step:2.13
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.9.2
    +-org.jenkins-ci.plugins.workflow:workflow-multibranch:2.26 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-multibranch:2.24
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.9.2
    +-org.jenkins-ci.plugins:pipeline-stage-step:2.5 (managed) <-- org.jenkins-ci.plugins:pipeline-stage-step:2.5
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.22
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.9.2
    +-org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.21 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-cps-global-lib:2.19
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkinsci.plugins:pipeline-model-definition:1.9.2
    +-org.jenkins-ci.plugins:pipeline-input-step:2.12 (managed) <-- org.jenkins-ci.plugins:pipeline-input-step:2.12
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.22
and
+-org.jenkins-ci.plugins.to-declarative:declarative-pipeline-migration-assistant:1.5.3-SNAPSHOT
  +-org.jenkins-ci.plugins:junit:1.53
    +-io.jenkins.plugins:checks-api:1.7.2 (managed) <-- io.jenkins.plugins:checks-api:1.7.2
      +-org.jenkins-ci.plugins.workflow:workflow-step-api:2.24 (managed) <-- org.jenkins-ci.plugins.workflow:workflow-step-api:2.23

```

Reply:
https://github.com/jenkinsci/bom/blob/1135.va_4eeca_ea_21c1/bom-2.289.x/pom.xml for example contains this version
(via ttps://github.com/jenkinsci/bom/blob/1135.va_4eeca_ea_21c1/bom-weekly/pom.xml#L22) 

dependency should be removed and bom should be updated